mesonbuild/modules/gnome.py

```diff
@@ -1333,17 +1333,21 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs: 'Y
             l_install_dir = os.path.join(install_dir, l, project_id)
 
             for i, m in enumerate(media):
+                m_file = None
                 m_dir = os.path.dirname(m)
                 m_install_dir = os.path.join(l_install_dir, m_dir)
+                try:
+                    m_file = mesonlib.File.from_source_file(state.environment.source_dir, l_subdir, m)
+                except MesonException:
+                    pass
```

Comment:
```suggestion
                try:
                    m_file: T.Optional[File] = mesonlib.File.from_source_file(state.environment.source_dir, l_subdir, m)
                except MesonException:
                    m_file = None
```

Makes a little more sense to me, but either way we need the annotation somewhere.

Otherwise from a technical point of view this looks fine to me, but I'm not really a gnome expert, I'm kinda taking your word on that this is what gnome wants.

Reply:
> Otherwise from a technical point of view this looks fine to me, but I'm not really a gnome expert, I'm kinda taking your word on that this is what gnome wants.

Adding @pwithnall (GNOME core module maintainer), so there is more clarity.